### PR TITLE
cql3: expr: Add unit tests for evaluate()

### PR DESCRIPTION
**Pull request by cvybhu**
This PR adds some unit tests for the `expr::evaluate()` function.

At first I wanted to add the unit tests as part of #11658, but their size grew and grew, until I decided that they deserve their own pull request.

I found a few places where I think it would be better to behave in a different way, but nothing serious.

### CODE REVIEW

File: test/boost/expr_test_utils.hh [hunk 1]
```diff
@@ -68,6 +68,115 @@ inline constant make_text_const(const sstring_view& text) {
...
+    std::vector<bytes> partition_key;
+    std::vector<bytes> clustering_key;
+    std::vector<managed_bytes_opt> static_and_regular_columns;
+    ::shared_ptr<selection::selection> selection;
+    query_options options;
+};
+using column_values = std::map<sstring, raw_value>;
+
+// Creates evaluation_inputs that can be used to evaluate columns and bind variables using evaluate()
+inline std::pair<evaluation_inputs, std::unique_ptr<evaluation_inputs_data>> make_evaluation_inputs(
```
Comment (avikivity) [inline test/boost/expr_test_utils.hh:316]: Such a large function shouldn't be inline, it increases compilation time.

**Review by avikivity — APPROVED** [review]
Looks good.

File: test/boost/expr_test_utils.hh (hunk 1 above)
Comment (cvybhu) [reply test/boost/expr_test_utils.hh:316]: At first I tried to add a matching .cc file and link it to the test, but I couldn't find an easy way to do it, so I decided that this is good enough.
Other header files in test/boost also have all functions `inline`, so I guess that's the convention there.

Comment (avikivity) [reply test/boost/expr_test_utils.hh:316]: There are plenty of .cc files in test/lib. Follow how one of them is referenced in configure.py.

**Comment by cvybhu** [comment]
v2:
* Split the header file into .hh and .cc, and put both of them in `test/lib`. Originally I tried to put the .cc file in `test/boost` and failed because of build problems.

File: test/boost/expr_test_utils.hh (hunk 1 above)
Comment (cvybhu) [reply test/boost/expr_test_utils.hh]: `test/lib` worked :)
done.

**Pull request closed without merging**